tests/integration/api/controllers/person.spec.js

```diff
@@ -104,4 +112,111 @@ describe('Person API', () => {
       });
     });
   });
+
+  describe('GET /api/v1/person', async () => {
+    const getPage = Person.v1.getPage(dataContext);
+    const limit = 100;
+    const cursor = '0';
+    const personType = 'person';
+    const invalidContactType = 'invalidPerson';
+    const onlineUserPlaceHierarchy = {
+      parent: {
+        _id: place1._id,
+        parent: {
+          _id: place2._id,
+        }
+      }
+    };
+    const offlineUserPlaceHierarchy = {
+      parent: {
+        _id: place0._id,
+        ...onlineUserPlaceHierarchy
+      }
+    };
+
+    it('returns a page of people', async () => {
+      const expectedPeople = [
+        contact0,
+        contact1,
+        contact2,
+        patient,
+        e2eTestUser,
+        {
+          type: personType,
+          ...userNoPerms.contact,
+          ...onlineUserPlaceHierarchy
+        },
+        {
+          type: personType,
+          ...offlineUser.contact,
+          ...offlineUserPlaceHierarchy
+        }
+      ];
+      const responsePage = await getPage(Qualifier.byContactType(personType), cursor, limit);
```

Comment:
```suggestion
      const responsePage = await getPage(Qualifier.byContactType(personType));
```

I don't want to get too deep into edge cases with these integration tests, but can we have this test cover the case where we use the default cursor/limit values.  Then add another test where we call `getPage` twice.  On the first call we give it a `null` cursor and a limit of `4`.  Then on the second call, we pass in the cursor value returned from the first call (and again a limit of `4`).  Then we can assert that between the two calls, we got the expected 7 persons back.  One of the key things I want to be careful to assert in the new test is the _value_ returned for the page cursor.  Should be `'4'` on the first call and then `null` on the second.

Reply:
One more case that I thought of that we should probably have is to call the `Person.v1.getAll` function and load up all the returned persons and make sure we get all 7 of them.  Just a safety check that our generator code integrates fine.

Reply:
> I don't want to get too deep into edge cases with these integration tests, but can we have this test cover the case where we use the default cursor/limit values. Then add another test where we call `getPage` twice. On the first call we give it a `null` cursor and a limit of `4`. Then on the second call, we pass in the cursor value returned from the first call (and again a limit of `4`). Then we can assert that between the two calls, we got the expected 7 persons back. One of the key things I want to be careful to assert in the new test is the _value_ returned for the page cursor. Should be `'4'` on the first call and then `null` on the second.

The reason I kept the integration test with the "non-paging" values for `limit` and `cursor` is that the objects being returned from the database are not in a consistent order for the order that we are passing into the `utils.saveDocs()` function. In our case, we pass `const allDocItems = [contact0, contact1, contact2, place0, place1, place2, patient];` into `utils.saveDocs` but it is not guaranteed that contacts will be returned in the same order. The order is undeterministic and random; it could be the order in which they were indexed into the view, so the page cannot be compared. For example, if I request a page with 4 items, it could be any 4 items.

Reply:
> The order is undeterministic and random; it could be the order in which they were indexed into the view

Minor, but I just want to clarify that order the data is _returned_ should not be random.  My understanding is that data returned from couch views is deterministically ordered based on the _key value_ and in cases like this where we have many results for the same key, the order is by `_id` value.  Each call to `getPage` with the same parameters, should return the same set of data (otherwise our skip-based paging would not work :sweat:).  However, this does not help us much for these integration tests since, as you have observed, the UUID's assigned as the `_id` value for our new contacts _are random_ and so, the order of the returned data could change between test runs.  

All that being said, I don't think we need to worry too much about trying to assert the _order_ of the data returned.  We don't even need to assert the _exact docs_ returned for each page.  For the test where we get multiple pages, it seems enough to just assert the size of the `data` array for each page.  Then at the end combine the `data` arrays into one array that contains all the contacts and do your `equalInAnyOrder` assertion to make sure that ultimately we got back all the contacts we expected.